doc/build/kconfig/extensions.rst

```diff
@@ -8,6 +8,42 @@ implementation of `Kconfig
 <https://docs.kernel.org/kbuild/kconfig-language.html>`__,
 which includes some Kconfig extensions:
 
+- Default values can be applied to existing symbols without
+  :ref:`weakening <multiple_symbol_definitions>` the symbols dependencies
+  through the use of ``configdefault``.
+
+  .. code-block:: none
+
+      config FOO
+          bool "FOO"
+          depends on BAR
+
+      configdefault FOO
+          default y if FIZZ
```

Comment:
The following are not equivalent:
```
     config FOO
          bool "FOO"
          default y
   
      config FOO
          default n
```
to
```
config FOO
          default n

     config FOO
          bool "FOO"
          default y
```

but is:
```
      config FOO
          bool "FOO"
          depends on BAR

      configdefault FOO
          default y if FIZZ
```
considered equivalent to:
```
      configdefault FOO
          default y if FIZZ

      config FOO
          bool "FOO"
          depends on BAR
```

Reply:
Playing with this PR, it seems they are not equivalent if multiple `defaults` exists, that is:
```
configdefault FOO
    default y if FIZZ

config FOO
    bool "FOO"
    default n
    depends on BAR
```

becomes:
```
config FOO
    bool "FOO"
    default y if FIZZ
    default n
    depends on BAR
```

but:
```
config FOO
    bool "FOO"
    default n
    depends on BAR

configdefault FOO
    default y if FIZZ
```

becomes:
```
config FOO
    bool "FOO"
    default n
    default y if FIZZ
    depends on BAR
```

Reply:
Correct, `default` ordering is preserved so that it is "as if" the `configdefault` node was a `config` with duplicated dependencies.

Reply:
I believe it is very important to highlight the exact behavior regarding a `configdefault` before and after the symbol.

It is likely that the use of `configdefault` will happen in a downstream Kconfig, meaning that it's likely to be in the middle of a Kconfig tree, for example like this:
```
# Sourced from a board Kconfig.defconfig file:
config FOO
    default y if BAR

# Sourced from a downstream Zephyr module Kconfig code:
configdefault FOO
    default n

# Sourced from the subsytem Kconfig code, which defines the original symbol:
config FOO
    bool "Main FOO symbol"
    depends on BAZ
```

Feel free to adjust above in the doc example, so that it suits what you believe is the most common use-case for users.